### PR TITLE
Added endpoint for all orders and HTTP client for deserialising all o…

### PR DESCRIPTION
**Pull request by mihai-m13**
Homework done

### CODE REVIEW

**Comment by mihai-m13** [comment]
Homework done

File: h2-service/src/main/java/dao/OrderDao.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package dao;
 
 import com.procourier.model.Order;
+import com.procourier.model.Product;
```
Comment (cribratu) [inline h2-service/src/main/java/dao/OrderDao.java]: Please remove unused imports and extra lines where appropriate

File: h2-service/src/main/java/main/MainApp.java [hunk 2]
```diff
@@ -1,12 +1,6 @@
 package main;
 
-import com.procourier.model.Address;
-import com.procourier.model.Buyer;
-import com.procourier.model.Courier;
-import com.procourier.model.Order;
-import com.procourier.model.OrderLine;
-import com.procourier.model.Product;
-import com.procourier.model.Seller;
+import com.procourier.model.*;
```
Comment (cribratu) [inline h2-service/src/main/java/main/MainApp.java]: You should not use * for import. It's better to import only the classes you need although they seem to be many

File: orders-service/src/main/java/com/procourier/app/OrderServiceMain.java [hunk 3]
```diff
@@ -42,11 +44,16 @@ public static void main(String[] args) throws Exception {
 
         Spark.port(8082);
         Spark.threadPool(26, 10, 2000);
-        //Spark.get("/orders/:id", handler);
-
-        Spark.get("/orders/:id", getOrderByIdHandler);
-
 
+        Spark.before((req, res) -> {
+            String path = req.pathInfo();
```
Comment (cribratu) [inline orders-service/src/main/java/com/procourier/app/OrderServiceMain.java]: Nice

File: orders-service/src/main/java/com/procourier/app/OrderServiceMain.java [hunk 4]
```diff
@@ -42,11 +44,16 @@ public static void main(String[] args) throws Exception {
...
+            if (path.endsWith("/"))
```
Comment (cribratu) [inline orders-service/src/main/java/com/procourier/app/OrderServiceMain.java]: Use brackets when using if/else although you have only a single line after it

File: orders-service-client/src/main/java/main/Main.java [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
 
 import com.procourier.model.Order;
 import http.GetOrderByIdClient;
+import http.GetOrdersClient;
+import java.util.List;
 
 public class Main {
     public static void main(String[] args) {
-        GetOrderByIdClient client = new GetOrderByIdClient();
+        GetOrderByIdClient byIdClient = new GetOrderByIdClient();
```
Comment (cribratu) [inline orders-service-client/src/main/java/main/Main.java]: Make all these variables final

File: orders-service-client/src/main/java/http/GetOrdersClient.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GetOrdersClient {
```
Comment (cribratu) [inline orders-service-client/src/main/java/http/GetOrdersClient.java]: You could have used the same client we created together

**Comment by cribratu** [comment]
Good job! 👍 No need to implement the changes, it was just an example of how things work on a real project.

**Comment by mihai-m13** [comment]
Thank you for all the feedback. Much appreciated.

**Pull request closed without merging**